src/accelerate/utils/other.py

```diff
@@ -31,6 +32,13 @@
     import torch_xla.core.xla_model as xm
 
 
+def is_compiled_module(module):
+    """Check whether the module was compiled with torch.compile()"""
+    if is_torch_version("<", "2.0.0") or not hasattr(torch, "_dynamo"):
+        return False
+    return isinstance(module, torch._dynamo.eval_frame.OptimizedModule)
```

Comment:
I'm guessing we don't have a public class we could use? This is going to get ugly if PyTorch decides to move this around.

Reply:
Agree, we had the same question in `diffusers`. Will also ask PyTorch about this

Reply:
https://huggingface.slack.com/archives/C047QHKH944/p1684250821270959

Reply:
Backwards comp will be kept, so think this is good for now. Merging.